include/itkBoneMorphometryFeaturesFilter.hxx

```diff
@@ -68,68 +66,43 @@ BoneMorphometryFeaturesFilter< TInputImage, TMaskImage >
   m_PlY = 0;
   m_PlZ = 0;
 
-  // Resize the thread temporaries
-  m_NumVoxelsInsideMask.SetSize(numberOfThreads);
-  m_NumBoneVoxels.SetSize(numberOfThreads);
-  m_NumX.SetSize(numberOfThreads);
-  m_NumY.SetSize(numberOfThreads);
-  m_NumZ.SetSize(numberOfThreads);
-  m_NumXO.SetSize(numberOfThreads);
-  m_NumYO.SetSize(numberOfThreads);
-  m_NumZO.SetSize(numberOfThreads);
-
-  // Resize the thread temporaries
-  m_NumVoxelsInsideMask.Fill(0);
-  m_NumBoneVoxels.Fill(0);
-  m_NumX.Fill(0);
-  m_NumY.Fill(0);
-  m_NumZ.Fill(0);
-  m_NumXO.Fill(0);
-  m_NumYO.Fill(0);
-  m_NumZO.Fill(0);
+  // Initialize atomics
+  m_NumVoxelsInsideMask.store(0);
+  m_NumBoneVoxels.store(0);
+  m_NumX.store(0);
+  m_NumY.store(0);
+  m_NumZ.store(0);
+  m_NumXO.store(0);
+  m_NumYO.store(0);
+  m_NumZO.store(0);
 }
 
 template< typename TInputImage, typename TMaskImage >
 void
 BoneMorphometryFeaturesFilter< TInputImage, TMaskImage >
 ::AfterThreadedGenerateData()
 {
-  ThreadIdType numberOfThreads = this->GetNumberOfThreads();
-
-  SizeValueType numVoxels = 0;
-  SizeValueType numBoneVoxels = 0;
-  SizeValueType numX = 0;
-  SizeValueType numY = 0;
-  SizeValueType numZ = 0;
-  SizeValueType numXO = 0;
-  SizeValueType numYO = 0;
-  SizeValueType numZO = 0;
-
-  for (unsigned int i = 0; i < numberOfThreads; ++i )
-    {
-    numVoxels += m_NumVoxelsInsideMask[i];
-    numBoneVoxels += m_NumBoneVoxels[i];
-    numX += m_NumX[i];
-    numY += m_NumY[i];
-    numZ += m_NumZ[i];
-    numXO += m_NumXO[i];
-    numYO += m_NumYO[i];
-    numZO += m_NumZO[i];
-    }
+  SizeValueType numVoxelsInsideMask = this->m_NumVoxelsInsideMask.load();
```

Comment:
`m_FieldName` convention was established to make it clear it is a member variable, so `this->` prefeix would not be required. Sometimes, `this->` is syntactically required. Is this one of those cases?